great_expectations/data_context/util.py

```diff
@@ -145,22 +146,29 @@ def substitute_config_variable(template_str, config_variables_dict):
     """
     if template_str is None:
         return template_str
+
     try:
-        match = re.search(r'^\$\{(.*?)\}$', template_str) or re.search(r'^\$([_a-z][_a-z0-9]*)$', template_str)
+        match = re.search(r'\$\{(.*?)\}', template_str) or re.search(r'\$([_a-z][_a-z0-9]*)', template_str)
     except TypeError:
         # If the value is not a string (e.g., a boolean), we should return it as is
         return template_str
 
+    ret = template_str
+
     if match:
         config_variable_value = os.getenv(match.group(1))
         if not config_variable_value:
             config_variable_value = config_variables_dict.get(match.group(1))
-            if not config_variable_value:
-                config_variable_value = template_str
-    else:
-        config_variable_value = template_str
 
-    return config_variable_value
+        if config_variable_value:
+                if match.start() == 0 and match.end() == len(template_str):
+                    ret = config_variable_value
+                else:
+                    ret = template_str[:match.start()] + config_variable_value + template_str[match.end():]
+
+    return ret
+
+
```

Comment:
Can this be simplified to:?

```suggestion
    # These two lines could even move up to substitute_all_config_variables, I think?
    vars = config_variables_dict
    vars.update(os.environ)
    return Template(template_str).safe_substitute(vars)
```

Reply:
ooh that's pretty slick James.

Reply:
@jcampbell @Aylr I love that this is so much shorter! Unfortunately, Template does not cover the case of the variable's value being anything other than a string. This is an issue for something like the credentials substitution with a dictionary.